### PR TITLE
HPCC-16710 Document Dali and Sasha data store roles

### PR DESCRIPTION
**Pull request by g-pan**
Fix HPCC-16710 Document Dali and Sasha data store roles
Added content revolving around the Backup, and High Availability section
not all being on the same node as the Thor master.

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>
@JamesDeFabia  FYI 
@RichardTaylorHPCC  please review
@jakesmith please review

### CODE REVIEW

**Comment by jakesmith** [comment]
Sure.

1) > Sasha works independently of yet in conjunction with Dali

Should there be a comma between "...of, yet..." ?

2) Sasha.

As I previously mentioned (in mast last reply below), this section doesn't mention the coalescing aspect.
Which is probably the most vital role Sasha plays.

So it should say that near the top of the list of jobs it does in this section.

3) > Allocating a larger disk for Sasha reduces the amount of housekeeping needed

A little misleading..
Increasing the disk size alone doesn't help.

What helps the environment (Dali) is if Sasha's archiving configuration is tuned such that Dali doesn't keep too many workunits in memory.
i.e. increasing the amount that Sasha archives, for which it needs plenty of disk to do so.

NB: Not sure you should explicitly mention, but there's no limit on the total Sasha archive size, e.g. if it's been running for years, all those archived workunits will still exist on the Sasha node unless ops. has deleted them or moved them somewhere.

4) > An Explicit request to save using dalidiag:

capitalization, should say "An explicit .."

5) > All Dali SDS changes are recorded in a delta transaction log (in XML format) named convention of daliincXXXX.xml

Should say : "All Dali SDS changes are recorded in a delta transaction log (in XML format), with a naming convention of daliincXXXX.xml

6) > Once the workunits are archived they are no longer available in the Dali data store. The archives can still be retrieved

Perhaps you should say, they can still be accessed via Eclwatch, by restoring them to Dali.

7) Multiple Nodes

It should be emphasised that putting any of the components on nodes with slaves on is bad and leads to an unbalanced cluster, resulting in those slaves with less memory/cpu taking longer than the rest and dragging the whole performance of the cluster down as a result.

**Comment by jakesmith** [comment]
@g-pan - a few comments

File: docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml [hunk 1]
```diff
@@ -175,34 +175,51 @@
...
 
           <para>Sasha also performs routine housekeeping such as removing
           cached workunits and DFU recovery files.</para>
+
+          <para>Sasha can also run XREF, to cross reference physical files
+          with logical metadata, to determine if there are lost/found/orphaned
+          files. It then presents options (via EclWatch) for their recovery or
+          deletion.</para>
+
+          <para>Sasha is the component responsible for removing expireds when
```
Comment (JamesDeFabia) [inline docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml]: expireds should be expired files

File: docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml [hunk 2]
```diff
@@ -527,17 +529,85 @@
...
+      HPCC's Dali processes are one of the few native active/passive
+      components. Using standard “swinging disk” clustering is recommended for
+      a high availability setup. For a single HPCC Dali process, any suitable
+      High Availability (HA) RAID level is fine.</para>
+
+      <para>Sasha only stores data to locally available disks, reading data
+      from Dali then processing it by archiving workunits (WUs) to disk. It is
+      beneficial to configure Sasha for a larger amount of archiving so that
+      Dali does not keep too many workunits in memory. This requires a larger
+      amount of diskspace. </para>
```
Comment (JamesDeFabia) [inline docs/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml]: diskspace s/b disk space.

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Minor changes requirede

**Comment by jakesmith** [comment]
@g-pan - changes look good.
But I see @JamesDeFabia has a couple of comments.

Back to you @g-pan - reassign when done.

**Comment by g-pan** [comment]
Requested changes made.  @jakesmith  please review.

**Review by jakesmith — APPROVED** [review]
Looks good.
@richardkchapman - ready to merge.

**Pull request merged**